modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -1191,8 +1189,13 @@ public String toInstantiationType(Schema p) {
 
     @Override
     public CodegenModel fromModel(String name, Schema model) {
+        LOGGER.debug("fromModel (schema): " + model);
```

Comment:
As a user with debug enabled, I'd have three concerns:

1. The use of debug calls in this method will cause a ton of noise as it'll be invoked with somewhat large object dumps every time for every model, potentially more than once per model
2. It's not clear from the log message what the difference is between `fromModel (schema):`, `fromModel (base end):`, and `fromModel (end):`.
3. The types being dumped are all different (Schema input, ADT before mutation, and ADT after mutation), which could be confusing especially if they have similar properties.

As a small point, the logging framework will interpolate content based on whether or not this log level is enabled using the `{}` formatter. Without this, a new string and model dump (which may or may not just be member reference string in this case, depends on the toString of the types) will be concatenated every time this line is hit regardless of whether or not debug is enabled.

```suggestion
        LOGGER.debug("fromModel (schema): {}", model);
```

Reply:
I wanted to leave these in as they were very helpful debugging issues, so to address your feedback:

1. I've downgraded these to trace
2 + 3. I've clarified the logs to make it clearer that the first is the schema, and the last is the model. I've eliminated the middle line.